### ISSUE TITLE
Deal with timeouts when connecting to multiple hosts

### ISSUE THREAD

**Issue by wellle** [aerospike/aerospike-client-go#151]
We noticed that sometimes `NewClientWithPolicyAndHost` returns an error if one of the provided hosts runs into a timeout. Here's a small example to illustrate:
```go
client, err := as.NewClientWithPolicyAndHost(as.NewClientPolicy(),
	as.NewHost("127.0.0.1", 4000), // times out
	as.NewHost("127.0.0.1", 3000), // works
)
if err != nil {
	log.Panicf("failed to connect: %s", err)
}
log.Printf("connected to %s", client)
```
For testing I have a local aerospike running in a VM on port 3000. When I run `nc -l 4000` to open port 4000 to time out and run the code above, then I see that it failed to connect. The same is true if I reorder both hosts. If I don't run `nc` or remove 4000 from the list of hosts, everything works.

This seems related to https://github.com/aerospike/aerospike-client-go/issues/41. And here, again, I feel like it should not fail if at least one of the hosts can be used to establish a connection.

So in this case it should find that 3000 does work, so lets connect to that cluster. Having other hosts in the list that are misbehaving should not make it fail.

What do you think?

Also, is there some option somewhere to have it retry on timeout? I did find there is one in `WritePolicy` after you're connected to a cluster. But there doesn't seem to be one in `ClientPolicy`. Would you think that would make sense to add?

**Comment by khaf** [comment]
Yeah, there have too many requests in this regard, maybe I should reconsider.

At the moment there is no retry bahavior defined in `ClientPolicy`. I'll also think about that and release a fix in the next release.

Thanks for your suggestions.

**Comment by khaf** [comment]
Christian, I just checked it appears that I have fixed this issue in the last few releases.

Could you check and see if you have the latest client?

**Comment by wellle** [comment]
@khaf: That's good to hear! We haven't updated recently, but are planning to. Do you happen to know in which version exactly this was fixed?

**Comment by khaf** [comment]
Should be v1.19.0. Your requested changes were implemented in v1.20.0 though.
